### DISCUSSION TITLE
Duplicate Statistics sections in Json

### DISCUSSION THREAD

**Discussion by quasiquote1** [sparcians/map#376] [Q&A]
Hi,

We're seeing a weird issue where sometimes there are two identical Statistics sections being printed in the JSON stats file:

```
{
  "Statistics":{
   }
}{
 "Statistics":{
   }
}
```

This is not exactly reproducible so not sure how to debug this issue.  Is this a known problem?  Thanks.

**Comment by klingaard** [comment]
Maybe you're seeing it during a time series generation?  I don't think I've seen it, but I might try reproducing it today.

**Reply by quasiquote1** [reply]
Not sure I understand what you mean by time series generation.  Can you explain what this means?

Do you think the same issue might arise if I use the report definition file?  So far I've only been using the auto report stats dumped at end of simulation (and specify JSON output).

**Reply by klingaard** [reply]
I tried both techniques: report definition file and a simple content file:

```
# report.def
content:
  report:
    def_file:  core_stats.yaml
    dest_file: out_stats.json
    format:    json
```
Used:
`./sparta_core_example --report report.def ...`
```
# core_stats.yaml
content:
  top:
    autopopulate:
      attributes: "!=vis:hidden"
```
Used:
`./sparta_core_example --report "" report.def out_report.json json`

and didn't see the duplication.

However, do you have 2 clocks in your simulation?  Maybe the report definition is supplied twice to the command line (with the same file)?  I dunno...  odd.

Timeseries info: https://sparcians.github.io/map/core_example.html  See "Generating Reports" (I need to add a TOC to this page) for an example of a time series report.  This is a report that is generated every `update-time <time>` or `update-count <stat value>` in CSV format (really the only format supported*).  Another example here: https://github.com/riscv-software-src/riscv-perf-model/blob/master/reports/core_timeseries.def as well as a brain-dead tool to observe it: https://github.com/riscv-software-src/riscv-perf-model/blob/master/reports/plot_ts.py

\* There is a SQLite3 format supported as well, but I haven't documented it yet.

**Comment by klingaard** [comment]
Did you ever figure this out/reproduce?

**Reply by quasiquote1** [reply]
Unfortunately not.  I have updated to the latest version of MAP so will see if this issue crops back up.
